AV-183081 Ingestion layer UT for Gateway

aaha97 opened the pull request:
This PR includes

1. added unit tests for gateway and gateway class (Create, Update and Delete)
2. object store for gateway and gateway class
3. gateway class validation. check if controller is set to AKO
4. code cleanup

aaha97 commented:
`=== RUN   TestGatewayCUD
2023-07-17T11:27:37.163+0530    INFO    utils/queue.go:121      Worker id 1
    gateway_test.go:127: Created gateway gw-example
    gateway_test.go:136: Updated gateway gw-example
    gateway_test.go:144: Deleted gateway gw-example
--- PASS: TestGatewayCUD (0.02s)
=== RUN   TestGatewaClassyCUD
    gateway_test.go:168: Created gateway class gw-class-example
    gateway_test.go:177: Updated gateway class gw-class-example
    gateway_test.go:185: Deleted gateway class gw-class-example
--- PASS: TestGatewaClassyCUD (0.02s)
PASS`

swathinsankaran commented on ako-gateway-api/tests/gateway_test.go:
```diff
@@ -0,0 +1,187 @@
+			Name: "gw-class-example",
+		},
+		Spec: gatewayv1beta1.GatewayClassSpec{
+			ControllerName: "ako.vmware.com/avi-lb",
```
Are we having a validation logic to ignore the gatewayclass if the controllerName is not given as `"ako.vmware.com/avi-lb"`? Shall we add a UT for that in the next PR?

swathinsankaran reviewed (approved):
LGTM

aaha97 commented:
`=== RUN   TestGatewayCUD
    gateway_test.go:127: Created gw-example
    gateway_test.go:136: Updated gw-example
    gateway_test.go:144: Deleted gw-example
--- PASS: TestGatewayCUD (0.02s)
=== RUN   TestGatewaClassyCUD
    gateway_test.go:168: Created gw-class-example
    gateway_test.go:178: Updated gw-class-example
    gateway_test.go:186: Deleted gw-class-example
--- PASS: TestGatewaClassyCUD (0.02s)
PASS`

swathinsankaran commented on ako-gateway-api/objects/gateway_store.go:
```diff
@@ -0,0 +1,83 @@
+	defer g.gwLock.Unlock()
+
+	_, gatewayList := g.GatewayClassToGateway.Get(gwClass)
+	return gatewayList.([]string)
```
The gatewayList item should contain `namespace/gatewayname` for easier mapping.

swathinsankaran commented on ako-gateway-api/objects/gateway_store.go:
```diff
@@ -0,0 +1,83 @@
+	//Gateway -> GatewayClass
+	GatewayToGatewayClass *objects.ObjectMapStore
+
+	//GatewayClass -> [gateway1, gateway2]
```
namespace also must be included since the gateway is a namespace scoped object.
